Add a blog post for the new terrainr update

mikemahoney218 opened the pull request:
Hi all! We briefly discussed on Twitter cross-posting the post I wrote for my own blog on the v 0.4.0 release of terrainr to the rOpenSci blog, so I'm opening this PR to do so! I've tagged the post for 2021-05-03 as that's a week from the next business day, but I'm of course open to changing that (and pretty much anything else in the post).

I believe I've done the following, please let me know if anything seems to be amiss:

* [ ] I have read the Content Guidelines.
* [ ] I have read the Technical Guidelines.
* [ ] I used or followed the R Markdown or Markdown template.
* [ ] I have followed the Style Guide.
* [ ] I created or updated my author metadata with correct folder name.
* [ ] I have added relevant tags after browsing existing tags (incuding "community" tag).
* [ ] I have added the "tech notes" tag if this is a technote.
* [ ] I ran `roblog::ro_lint_md()` on index.md (optional).
* [ ] I ran `roblog::ro_check_urls()` on index.md (optional).
* [ ] I ran a spell-check on index.md.
* [ ] I have added the tags - Software Peer Review, my-packagename.
* [ ] I have added the package-version YAML tag.
* [ ] I have added acknowledgement of the reviewers' work (with links to reviewers).
* [ ] I have added a link to the software peer review thread.
* [ ] I ran a spell-check on index.md.

steffilazerte commented:
Awesome! I'll take a look next week and @stefaniebutland will give us a publication date.

stefaniebutland commented:
I think 2021-05-04 (Tues May 4) to publish, since community call summary should be published week of April 26th.

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+
+{{< figure src = "washington.jpg" alt = "A 3D rendering of the summit of Mt. Washington, including watershed boundaries and river lines." >}}
+
+## `merge_rasters` can handle tiles with different numbers of bands
```
This is a super persnickity suggestion :grin:
Above you use `update.packages()` for functions and here you use `merge_rasters` (no braces) for a function name. We don't have a guideline for which format is best, but I'd suggest being consistent with either braces or no braces. 

My personal preference is for braces (helpful for new users to remember that functions almost always need them!) but in this case, the easiest fix is just to remove them from `update.packages()` :grin: 

So which ever you prefer is fine!

mikemahoney218 replied:
I took the easy way out :laughing: 

Later in the post I mention that `gdalwarp` throws some errors when trying to merge rasters with differing numbers of bands, but `gdalwarp` isn't actually an R function (I think it's implemented in Python, but it's called via GDAL which is called via sf) which causes a weird situation -- does a function that you can't call from R get `()` like every other function, or does it not get `()` and then be formatted like a package? Your suggestion of dropping them from `update.packages` helps me avoid that :laughing:

steffilazerte replied:
Totally your prerogative! That's a good point about mixing functions between packages and I know rOpenSci doesn't have a recommendation for functions (these odd scenarios are probably why! Easiest to adapt to what works best).  

For me, because I'm usually writing R documentation for people are less familiar with R, I use `function()` simply because I don't have other non-R functions to get in the way, and I'm trying really, really, really hard to be explicit about what's a function and what's the name of an object (like a data frame or argument in an example). So that's a different situation :grin:

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+tiles into memory at once, and was a bit of a mess to maintain thanks to the 
+large number of paths you could theoretically take through the code. The commit
+(suggested via rOpenSci review!) replacing it with `gdalwarp` via `sf` is 
+probably the single best code improvement I've made to this repo. Unfortunately,
```
```suggestion
probably the single best code improvement I've made to this package. Unfortunately,
```

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+elevation data and basemap tiles for points within the United States (using 
+public domain data from the USGS National Map), visualize them in R via ggplot2,
+and process them for 3D visualization using the Unity 3D engine. You can see
+[the GitHub repo here!](https://docs.ropensci.org/terrainr/)
```
Technically the link is to the docs page (I'm being picky!)

```suggestion
[the docs and access the GitHub repo here!](https://docs.ropensci.org/terrainr/)
```

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+title: What's New in terrainr 0.4.0?
+author:
+  - Mike Mahoney
+date: '2021-05-03'
```
**Also** Remember to change the folder name!
```suggestion
date: '2021-05-04'
```

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+  - tech notes
+package_version: 0.4.0
+description: The new CRAN release of terrainr improves consistency, CRS logic, and fixes some bugs.
+twitterImg: blog/2021/05/03/what-s-new-in-terrainr-0-4-0/washington.jpg
```
```suggestion
twitterImg: blog/2021/05/04/what-s-new-in-terrainr-0-4-0/washington.jpg
```

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+  - data-retrieval
+  - ggplot2
+  - geospatial
+  - peer-reviewed
```
```suggestion
  - Software Peer Review
```

steffilazerte commented on content/blog/2021-05-03-what-s-new-in-terrainr-0-4-0/index.Rmd:
```diff
@@ -0,0 +1,149 @@
+The old implementation of `merge_rasters` was very bulky, read all your map 
+tiles into memory at once, and was a bit of a mess to maintain thanks to the 
+large number of paths you could theoretically take through the code. The commit
+(suggested via rOpenSci review!) replacing it with `gdalwarp` via `sf` is 
```
Since this is your first post about terrainr, it'd be great to acknowledge the reviewers (perhaps at the end?) and here, you could link to the rOpenSci review issue.

mikemahoney218 replied:
Added to the "What's a terrainr" header :)

steffilazerte replied:
Great!

steffilazerte reviewed (commented):
Hi @mikemahoney218 

That was super speedy from tweet to post! This is a nice tech note to update people on terrainr: short and sweet with all the important points and definitely useful information to share. 

I have just a couple of  suggestions (7) most of which are persnickety styling things and a couple related to the publication date change and the fact that as this is the first blog post on rOpenSci about terrainr, it'd be great to acknowledge the reviewers and software review process. 

We're good to go with Tuesday May 4th for publication, just remember to change the date in the YAML as well as folder names and links.

Pretty small revisions so if I could get your edits by Friday or Monday, we should be good, Thanks!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [x] publication date is ok
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok ("tech notes" for tech notes; "community" for non-staff non-editor)
* [X] YAML package-version included
* [x] YAML subject tags - software peer review, packagename
* [x] acknowledges and links to reviewers
* [ ] links to peer review thread

mikemahoney218 commented:
Thanks for the comments @steffilazerte ! I got an email from GitHub while on lunch and the edits seemed easy enough, so I went through and made them now.

I believe I've addressed all the comments, let me know if there's anything to still be changed. Thanks!

steffilazerte commented:
Awesome! Thanks for a super speedy review, we're ready to publish on next Tuesday. I'll post a draft tweet here soon. Feel free to comment on it if you have suggestions, otherwise, we're done!

steffilazerte commented:
Draft tweet:

------------------

[technote] What's New in terrainr 0.4.0? 🏔️⛰️

@&#8203;mikemahoney218 explains updates to terrainr

- merge_rasters handles tiles with different no. bands
- get_tiles no more transparency auto-download for NAIP imagery
- And more!

https://ropensci.org/blog/2021/05/04/what-s-new-in-terrainr-0-4-0/

#rstats #terrainr
(1/2)

---

🙏 And of course, big thanks to rOpenSci reviewers @&#8203;__mike_johnson & Sydney Folks (https://github.com/sfoks)

(2/2)

mikemahoney218 commented:
Looks good @steffilazerte ! I'd possibly suggest:

* Functions infer CRS from provided sf and Raster objects

As the second bullet; I think the NAIP change is subtle enough that it's hard to explain in a tweet amount of space. But fine either way!

steffilazerte commented:
Good point. Edited!

> Looks good @steffilazerte ! I'd possibly suggest:
> 
>     * Functions infer CRS from provided sf and Raster objects
> 
> 
> As the second bullet; I think the NAIP change is subtle enough that it's hard to explain in a tweet amount of space. But fine either way!
